### PR TITLE
Reg 8 - Create Major Schema

### PR DESCRIPTION
**Pull request by arianaolson419**
https://olinjs.atlassian.net/browse/REG-8

This is a very simple major schema that has keys for the name, general requirements (stuff everyone needs to graduate), and major requirements (major-specific things). Currently, the schema defaults to Mechanical Engineering values. Sources for requirements: http://www.olin.edu/printpdf/course-catalog/program-specific-graduation-requirements/ (Major reqs) http://www.olin.edu/printpdf/course-catalog/general-graduation-requirements/ (General reqs). Defaults may be changed/taken away in the future, but for now, it allows us to do some testing and have a concrete idea of what a complete major object should look like.

### CODE REVIEW

File: models/majorModel.js
Comment (billmwong) [inline models/majorModel.js]: Same thing I told Logan, this comment isn't really necessary here

Comment (billmwong) [inline models/majorModel.js]: same thing I told Logan here, this comment isn't really necessary because I can tell what you're doing here from the code.

**Review by billmwong — CHANGES_REQUESTED** [review]
Nice documentation, nice es6! The content all looks good, but the style is not up to the linter standards. Make sure all your indents are using 2 spaces instead of tabs, and a couple other linter issues I saw.

I'm working on making the linter for the backend code run automatically, but for now run the following in the terminal to test this file on the linter:
```
./node_modules/.bin/eslint models/majorModel.js
```

**Review by billmwong — APPROVED** [review]
Nice, looks good from me now! Waiting on a student review before merging.

File: models/majorModel.js
Comment (logandavis) [inline models/majorModel.js]: nit: there is only one "l" in "Modeling"

File: models/majorModel.js
Comment (logandavis) [inline models/majorModel.js]: nit: should be "Physics Foundation" to match biology and AHS

File: models/majorModel.js
Comment (logandavis) [inline models/majorModel.js:21]: nit: I think there should be spaces on either side of the slash

File: models/majorModel.js
Comment (logandavis) [inline models/majorModel.js]: so, I took ModCon and RWM instead of ISIM. is the plan to just say that those fulfill the "ISIM" requirement when creating those courses in the database?

**Review by logandavis — CHANGES_REQUESTED** [review]
Mostly looks good, some nitpicks in the schema defaults.

File: models/majorModel.js
Comment (robbie-sundstrom) [reply models/majorModel.js]: I think the cleanest way to do this might be to have two separate requirements, one for ModCon and one for RWM. Then ISIM can just fulfill both of those requirements.

Comment (arianaolson419) [reply models/majorModel.js]: Yeah that is definitely something to think about. For now I just copied the document I was referencing gas closely as possible. Bill talked to Linda about how they handle stuff like changing requirements now, so we might get some inspiration from that.

File: models/majorModel.js
Comment (arianaolson419) [reply models/majorModel.js]: Yeah I thought about that, but I decided to match the document I was referencing as closely as possible. Looking back at it, I agree that Physics is kinda vague and gross. I'll change it to Physics Foundation

**Review by logandavis — APPROVED** [review]

**Comment by billmwong** [comment]
@arianaolson419 Resolve those merge conflicts and then I can merge!

File: models/majorModel.js
Comment (billmwong) [inline models/majorModel.js]: Ahh don't get rid of this line

**Review by billmwong — CHANGES_REQUESTED** [review]

**Review by billmwong — APPROVED** [review]

**Pull request merged**
